### PR TITLE
[Task 1872109] Migrate to use web app create/deploy task in web app maven plugin

### PR DESCRIPTION
**Pull request by Flanker32**
Migrate to use shared task in library and refactor maven mojo to follow following step:
  1. Tooling validator validate raw configuration(pom configuration in maven)
  2. Parser get task config from raw configuration
  3. Create correspond task with config and execute
Repeat 2 3 to complete all tasks

[AB#1872109](https://dev.azure.com/mseng/a4d27ce2-a42d-4b71-8eef-78cee9a9728e/_workitems/edit/1872109)

### CODE REVIEW

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java [hunk 1]
```diff
@@ -238,11 +238,11 @@ protected void validateConfiguration(Consumer<ValidationMessage> validationMessa
         }
     }
 
-    protected synchronized WebAppConfig getWebAppConfig() throws AzureExecutionException {
-        if (config == null) {
-            config = new ConfigParser(this).parse();
+    public ConfigParser getConfigParser() {
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/AbstractWebAppMojo.java:241]: create parser directly inline

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/ConfigParser.java [hunk 2]
```diff
@@ -135,71 +199,40 @@ public Runtime getRuntime() {
...
-                .region(getRegion())
-                .runtime(getRuntime())
-                .dockerConfiguration(getDockerConfiguration())
-                .deploymentSlotName(getDeploymentSlotName())
-                .deploymentSlotConfigurationSource(getDeploymentSlotConfigurationSource())
-                .webAppArtifacts(getMavenArtifacts())
-                .appSettings(this.mojo.getAppSettings())
-                .build();
+    private RuntimeConfig getRuntimeConfig() throws AzureExecutionException {
+        final MavenRuntimeConfig runtime = mojo.getRuntime();
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/parser/ConfigParser.java:214]: handle os == null, we may need to parse windows/linux/docker

File: azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java [hunk 3]
```diff
@@ -6,293 +6,110 @@
...
+        new DeployWebAppTask(target, artifacts, isStopAppDuringDeployment()).execute();
     }
 
-    private void deployArtifactsWithZipDeploy(IWebAppBase target, List<WebAppArtifact> artifacts) throws AzureExecutionException {
-        final File stagingDirectory = prepareStagingDirectory(artifacts);
-        // Rename jar once java_se runtime
-        if (Objects.equals(target.getRuntime().getWebContainer(), WebContainer.JAVA_SE)) {
-            final List<File> files = new ArrayList<>(FileUtils.listFiles(stagingDirectory, null, true));
-            DeployUtils.prepareJavaSERuntimeJarArtifact(files, project.getBuild().getFinalName());
+    private void deployExternalResources(final IAppService<?> target, final List<DeploymentResource> resources) throws AzureExecutionException {
```
Comment (Flanker32) [inline azure-webapp-maven-plugin/src/main/java/com/microsoft/azure/maven/webapp/DeployMojo.java:91]: Extract to a task in maven tooling

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**